### PR TITLE
Bump up to node 18

### PR DESCRIPTION
**Pull request by DmitryLukyanov**
## Description

Bump up used node version to 18 and fixing some warnings happened during build and test scripts

## Type of changes

Migration to Node 18.

## Checklist

- [x] I have read [CONTRIBUTING](https://github.com/epam/miew/blob/master/CONTRIBUTING.md) and [CODE_OF_CONDUCT](https://github.com/epam/miew/blob/master/CODE_OF_CONDUCT.md) guides.
- [ ] I have followed the code style of this project.
- [x] I have run `yarn run ci`: lint and tests pass locally with my changes.
NOTE: Additionally I've launched: `build`, `test` and `serve` scripts from the root. 
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation / The changes do not need docs update.

Some notes:

1. There is a number of different warnings during building that might be better to fix. 
2. There is no script: `yarn run demo` that is suggested to run in `CONTRIBUTING.md`.
3. Should running something like `npx prettier --write .` be requited before merging? At some point, I saw failed tests that required it, but now for some reason all has been built smoothly without it, so I reverted that changes, but what is regular workflow regarding this styling question?
4. I see that `*.min.js` files are under the git, should they be excluded?
5. There are at least new version for `eslint*` packages, does it make sense to update them too?
cc: @NataliaDSmirnova , @paulsmirnov

### CODE REVIEW

File: demo/webpack.config.mjs [hunk 1]
```diff
@@ -7,7 +7,11 @@ import HtmlWebpackPlugin from 'html-webpack-plugin'
 import ScriptExtHtmlWebpackPlugin from 'script-ext-html-webpack-plugin'
 import MiniCssExtractPlugin from 'mini-css-extract-plugin'
 import CopyWebpackPlugin from 'copy-webpack-plugin'
-import version from './tools/version'
+import version from './tools/version.mjs'
```
Comment (DmitryLukyanov) [inline demo/webpack.config.mjs:10]: I've tried to proceed using `webpack.config.babel.js` and `version.js`, but with no luck, thanks @NataliaDSmirnova for the help here

**Comment by paulsmirnov** [comment]
The `master-alt` branch has a nice addition that can also work for this PR - force and check the Node/NPM version
- https://github.com/epam/miew/blob/master-alt/.npmrc
- https://github.com/epam/miew/blob/master-alt/package.json#L10

**Comment by NataliaDSmirnova** [comment]
@DmitryLukyanov Great job! Thank you for the important step of making the project up-to-date.

@paulsmirnov, could you comment on the suggestions from the PR description?
As for me:
[1-3, 5] I suggest creating tickets, but where? :)
[4] I really don't know. Let's talk about it.

**Review by paulsmirnov — APPROVED** [review]
I'm not sure if we need to commit changes in `dist` but it seems that the project is currently configured to do so. Hmm. Let's check and update the configuration as a separate task. Originally, we updated dist files only during releases.

**Comment by NataliaDSmirnova** [comment]
> I'm not sure if we need to commit changes in `dist` but it seems that the project is currently configured to do so. Hmm. Let's check and update the configuration as a separate task. Originally, we updated dist files only during releases.

An issue was added https://github.com/epam/miew/issues/439

**Pull request merged**